components/logging/org.wso2.carbon.logging.appender.http/src/main/java/org/wso2/carbon/logging/appender/http/SecuredHttpAppender.java

```diff
@@ -356,21 +399,80 @@ private String resolveSecretPassword(String password) {
         return password;
     }
 
+    private void printWarningLogOnRemoteServerFailure() {
+
+        long diskUsage = persistentQueue.getCurrentDiskUsage();
+        switch (this.failureWarningLevel) {
+            case NONE:
+                getStatusLogger().warn("Remote log publishing failure : Unable to publish logs to the remote server. " +
+                        "Please check the remote server status.");
+                this.failureWarningLevel = FailureWaringLevel.INITIAL;
+                break;
+            case INITIAL:
+                if (diskUsage > persistentQueue.getMaxDiskSpaceInBytes() / 2) {
+                    getStatusLogger().warn("Remote log publishing failure : The number of logs in the queue has exceeded" +
+                            " 50% of the allocated queue limit. Please check the remote server status.");
```

Comment:
check usage from disk read here

Reply:
The disk usage check is moved to the queue block creation and deletion methods. Therefore the value kept on memory on persistentQueue is always accurate. Therefore not need to check here.